### PR TITLE
Implement initial functionality

### PR DESCRIPTION
**Pull request by m1j09830**
## 🗣 Description ##

This PR contains the code to create an AMI with the [Egress-Assess tool](https://github.com/tarrell13/Egress-Assess/tree/Forked) software.

## 💭 Motivation and context ##

The Egress-Assess AMI will be used for egress testing of networks during engagements.

## 🧪 Testing ##

An AMI was built, deployed, and successfully tested in a staging environment.

Tests were conducted to ensure proper connection and exchange of data between an Egress-Assess client and the server running in the staging environment.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Build final pre-release AMI and test
- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

File: src/packer.pkr.hcl [hunk 1]
```diff
@@ -59,15 +59,15 @@ data "amazon-ami" "debian_bullseye" {
 
 locals { timestamp = regex_replace(timestamp(), "[- TZ:]", "") }
 
-source "amazon-ebs" "example" {
+source "amazon-ebs" "egress-assess" {
```
Comment (jsf9k) [inline src/packer.pkr.hcl]: We prefer snake case for HCL:
```suggestion
source "amazon-ebs" "egress_assess" {
```

File: src/packer.pkr.hcl [hunk 2]
```diff
@@ -110,7 +110,7 @@ source "amazon-ebs" "example" {
 }
 
 build {
-  sources = ["source.amazon-ebs.example"]
+  sources = ["source.amazon-ebs.egress-assess"]
```
Comment (jsf9k) [inline src/packer.pkr.hcl]: To maintain consistency with my previous suggestion in this file:
```suggestion
  sources = ["source.amazon-ebs.egress_assess"]
```

File: README.md [hunk 3]
```diff
@@ -1,14 +1,9 @@
...
 
-This is a generic skeleton project that can be used to quickly get a
-new [cisagov](https://github.com/cisagov) GitHub
-[Packer](https://packer.io) project started. This skeleton project
-contains [licensing information](LICENSE), as well as
-[pre-commit hooks](https://pre-commit.com) and
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This project can be used to build a Debian AMI with
+[Egress-Assess](https://github.com/FortyNorthSecurity/Egress-Assess) installed.
```
Comment (jsf9k) [inline README.md]: You're installing from [tarrell13/Egress-Assess](https://github.com/tarrell13/Egress-Assess), and according to #13 you intend to continue doing so, so this link should be updated:
```suggestion
[Egress-Assess](https://github.com/tarrell13/Egress-Assess) installed.
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
Looks good!  I'm only requesting a few minor changes.

**Review by jsf9k — APPROVED** [review]
:+1:

**Review by dav3r — APPROVED** [review]
Great work @m1j09830 🚀

**Comment by jsf9k** [comment]
@m1j09830 - don't forget to finalize the version before merging.

**Comment by dav3r** [comment]
@mcdonnnj - Please review this PR even though it already has 2 dev reviews, since I helped @m1j09830 significantly and therefore my review is fairly biased.

File: src/cloud_init.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  tasks:
+    - name: Enable all cloud-init services
+      ansible.builtin.service:
+        name: "{{ item }}"
+        enabled: yes
+      loop:
+        - cloud-init-local
+        - cloud-init
+        - cloud-config
+        - cloud-final
```
Comment (mcdonnnj) [inline src/cloud_init.yml]: We prefer to sort things like this. If the current order is required then it should be documented. I see this appears to be an inherited thing from cribbing other Packer projects so as long so we should update those projects based on the resolution of this conversation.

```suggestion
        - cloud-config
        - cloud-final
        - cloud-init
        - cloud-init-local
```

File: src/egress_assess.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  name: Install Egress-Assess tool
+  become: yes
+  become_method: sudo
+  roles:
+    - role: assessment_tool
+      vars:
+        # TODO: Switch to master branch after Forked branch is merged
+        # See https://github.com/cisagov/egress-assess-packer/issues/13
+        archive_src: "https://github.com/tarrell13/Egress-Assess/\
+        tarball/Forked"
```
Comment (mcdonnnj) [inline src/egress_assess.yml]: This needs to be updated in line with the resolution of https://github.com/cisagov/kali-packer/pull/141#discussion_r1089255813.

File: README.md [hunk 6]
```diff
@@ -1,14 +1,9 @@
...
 
-This is a generic skeleton project that can be used to quickly get a
-new [cisagov](https://github.com/cisagov) GitHub
-[Packer](https://packer.io) project started. This skeleton project
-contains [licensing information](LICENSE), as well as
-[pre-commit hooks](https://pre-commit.com) and
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This project can be used to build a Debian AMI with
+[Egress-Assess](https://github.com/tarrell13/Egress-Assess) installed.
```
Comment (mcdonnnj) [inline README.md]: This also needs to be updated with the resolution of https://github.com/cisagov/kali-packer/pull/141#discussion_r1089255813.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: src/cloud_init.yml (hunk 4 above)
Comment (dav3r) [reply src/cloud_init.yml]: Yes, this list of items should be sorted here and elsewhere.

File: src/egress_assess.yml (hunk 5 above)
Comment (m1j09830) [reply src/egress_assess.yml]: Made changes in `egress_assess.yml` to reflect updated repository. Please see f877fb21591423a51ef16ce07bb25397fcd1d0d3. Updated issue #13 to properly reflect the changes and close out the issue.

File: README.md (hunk 6 above)
Comment (m1j09830) [reply README.md]: Made the changes to the README per the suggestion above. Please see 14c06dc77fda40c87e7bfde16ff90cf7531b6efe

File: terraform-build-user/main.tf [hunk 7]
```diff
@@ -9,6 +9,13 @@ module "iam_user" {
     aws.images-staging-ssm    = aws.images-staging-ssm
   }
 
-  ssm_parameters = ["/cyhy/dev/users", "/ssh/public_keys/*"]
-  user_name      = "build-egress-assess-packer"
+  ssm_parameters = [
+    "/cyhy/dev/users",
+    "/ssh/public_keys/*",
```
Comment (mcdonnnj) [inline terraform-build-user/main.tf]: I don't believe this configuration needs access to these SSM parameters.
```suggestion
```

File: src/egress_assess.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+---
+- hosts: all
+  name: Install Egress-Assess tool
```
Comment (mcdonnnj) [inline src/egress_assess.yml]: I'm fine with the playbook name remaining the same but I think this descriptor should reflect the tool being installed.

```suggestion
  name: Install Auto-Egress-Assess tool
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Thanks for resolving my previous feedback. I have two more small items from re-reviewing.

File: terraform-build-user/main.tf (hunk 7 above)
Comment (dav3r) [reply terraform-build-user/main.tf]: Ah you are right now that we have stopped including [cisagov/ansible-role-dev-ssh-access](https://github.com/cisagov/ansible-role-dev-ssh-access) in our AMIs.  We should probably remove these [from skeleton-packer also](https://github.com/cisagov/skeleton-packer/blob/078fd4c8cbaac9597a3e599f6b9f487b54a4b5b2/terraform-build-user/main.tf#L12).

Comment (mcdonnnj) [reply terraform-build-user/main.tf]: @dav3r Yes I have this draft PR toward that work: https://github.com/cisagov/ami-build-iam-user-tf-module/pull/53

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

**Pull request merged**